mrmustard/lab_dev/states/gaussian_state.py

```diff
@@ -76,22 +74,22 @@ def __init__(
                 update_symplectic,
             )
         )
-
         self._representation = self.from_ansatz(
             modes=modes,
             ansatz=PolyExpAnsatz.from_function(
                 fn=triples.gket_state_Abc, symplectic=self.parameters.symplectic
             ),
         ).representation
 
-    def _getitem_builtin(self, modes: set[int] | Sequence[int]):
+    def __getitem__(self, idx: int | Collection[int]):
         r"""
-        The slicing method for a GDM state.
+        Override the default __getitem__ method to handle symplectic slicing.
         """
-
-        remaining_modes = [mode for mode in self.modes if mode not in modes]
-
-        return self >> TraceOut(remaining_modes)
+        idx = (idx,) if isinstance(idx, int) else idx
+        if not set(idx).issubset(set(self.modes)):
+            raise ValueError(f"Expected a subset of ``{self.modes}``, found ``{idx}``.")
+        trace_out_modes = set(self.modes) ^ set(idx)
```

Comment:
this is a bit risky because you might get a result with scrambled modes. `[mode for mode in self.modes if mode not in modes]` preserves the order.